packages/flutter/lib/src/services/text_input.dart

```diff
@@ -16,47 +16,118 @@ export 'dart:ui' show TextAffinity;
 /// The type of information for which to optimize the text input control.
 ///
 /// On Android, behavior may vary across device and keyboard provider.
-enum TextInputType {
+///
+/// This class allows additional flags for some input types. For example,
+/// numeric input can specify whether it supports decimal numbers and/or
+/// signed numbers.
+class TextInputType {
+  const TextInputType._(this.type) : signed = null, decimal = null;
+
+  /// Optimize for textual information.
+  ///
+  /// Requests a numeric keyboard with additional settings.
+  /// The [signed] and [decimal] parameters are optional.
+  const TextInputType.numberWithOptions({
+    this.signed: false,
+    this.decimal: false,
+  }) : type = 2;
+
+  final int type;
+
+  /// The number is signed, allowing a positive or negative sign at the start.
+  /// This flag is only used for numeric input type.
```

Comment:
put a blank line between the sentences, so that only the first is considered part of the "brief" description